### PR TITLE
arrays-creating_arrays-hard-q1

### PR DESCRIPTION
**Pull request by SamamthaH**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/arrays/creating_arrays/hard/q1/RandomArray.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package arrays.creating_arrays.hard.q1;
+
+/*
+Write a Java program that asks the user for the number of elements it will store and uses a method called
+createRandomArray() to create a new integer array and initializes the array with random numbers from 0 to the number
+that was entered. The main method should then print out the numbers in the array.
+Sample output:
```
Comment (Hedgemon4) [inline src/arrays/creating_arrays/hard/q1/RandomArray.java:7]: Add space between question text and sample output

File: src/arrays/creating_arrays/hard/q1/RandomArray.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+Enter the size of the array:
+5
```
Comment (Hedgemon4) [inline src/arrays/creating_arrays/hard/q1/RandomArray.java:9]: Tab this over

File: src/arrays/creating_arrays/hard/q1/RandomArray.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package arrays.creating_arrays.hard.q1;
+
+/*
+Write a Java program that asks the user for the number of elements it will store and uses a method called
```
Comment (Hedgemon4) [inline src/arrays/creating_arrays/hard/q1/RandomArray.java:4]: Change question text to:
Write a Java program that asks the user to enter a length for an array and then calls a method named createRandomArray() to generate this array. The method should accept an integer for the length and create a new array of that size. Then, it should initialize the array with random numbers from 0 (inclusive) to the passed length (exclusive). Return this array and print the output in the main method.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Comment by SamamthaH** [comment]
I think the reason this is framework is because without the try catches we still don't have a good way of checking if the entire array is random ints

File: src/arrays/creating_arrays/hard/q1/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        RandomArray.main(new String[0]);
+    }
+
+    public static Stream<Integer> createRandomArrayInputProvider() {
+        return Stream.of(1000, 1500, 2000, 3000);
```
Comment (Hedgemon4) [inline src/arrays/creating_arrays/hard/q1/MainTest.java:45]: We should probably try some other values that are not so large. With larger values, it is more likely they can pass the random check without actually doing it properly.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/creating_arrays/hard/q1/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    @MethodSource("createRandomArrayInputProvider")
+    public void correctCreateRandomArrayMethod(int in) throws Throwable {
+        Object[][] arguments = {
+                {in, int.class}
+        };
+        MethodTest m = new MethodTest(RandomArray.class, "createRandomArray", arguments);
+        Object output = m.callMethod();
+        try {
+            int[] arr = (int[]) output;
+            ArrayList<Integer> response = Arrays.stream(arr).boxed().collect(Collectors.toCollection(ArrayList::new));
```
Comment (Hedgemon4) [inline src/arrays/creating_arrays/hard/q1/MainTest.java:62]: We should also verify that the array returned is the correct length

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**